tpdatasrc/tpgamefiles/scr/tpModifiers/sp_magic_circle_outward_fixed.py

```diff
@@ -0,0 +1,231 @@
+]
+
+#Checks if a character's alignment matches the spell type
+def CheckAlignment(character, flag):
+	
+	align = character.stat_level_get(stat_alignment)
+	if flag == 1:
+		return align & ALIGNMENT_GOOD
+	elif flag == 2:
+		return align & ALIGNMENT_EVIL
+	elif flag == 3:
+		return align & ALIGNMENT_LAWFUL
+	elif flag == 4:
+		return align & ALIGNMENT_CHAOTIC
+
+	return 0
+	
+#Check if a summoned creature can still attack someone protected
+def SummonCanAttack(character, flag):
+	
+	align = character.stat_level_get(stat_alignment)
+	if flag == 1:
+		return align & ALIGNMENT_EVIL
+	elif flag == 2:
+		return align & ALIGNMENT_GOOD
+	elif flag == 4:
+		return align & ALIGNMENT_CHAOTIC
+	elif flag == 3:
+		return align & ALIGNMENT_LAWFUL
+
+	return 0
+
+#Gets the text name for the type of spell
+def GetSpellType(type):
+	TypeName = ""
+	if type == 1:
+		TypeName = "Good"
+	elif type == 2:
+		TypeName = "Evil"
+	elif type == 3:
+		TypeName = "Law"
+	elif type == 4:
+		TypeName = "Chaos"
+	return TypeName
+
+def MagicCircleOutwardBegin(attachee, args, evt_obj):
+	radius_feet = 10.0
+	obj_evt_id = attachee.object_event_append(OLC_CRITTERS, radius_feet)
+	args.set_arg(3, obj_evt_id)
+	print "Magic Circle Outward: New Object Event ID: " + str(obj_evt_id)
+	return 0
+
+def MagicCircleOutwardAoEEntered(attachee, args, evt_obj):
+	obj_evt_id = args.get_arg(3)
+
+	if obj_evt_id != evt_obj.evt_id:
+		print "Magic Circle Outward Aura Entered: ID mismatch " + str(evt_obj.evt_id) + ", stored was: " + str(obj_evt_id)
+		return 0
+
+	print "Magic Circle Outward Aura Entered, event ID: " + str(obj_evt_id)
+	tgt = evt_obj.target
+	if tgt == OBJ_HANDLE_NULL:
+		return 0
+	if attachee == OBJ_HANDLE_NULL:
+		return 0
+
+	#All get the effect even the character the spell is on
+	type = args.get_arg(2)
+	tgt.condition_add_with_args("Magic Circle Outward Aura", obj_evt_id, type, 0)
+	return 0
+	
+def MagicCircleOutwardHasSpellActive(attachee, args, evt_obj):
+	evt_obj.return_val = 1
+	return 0
+	
+def MagicCircleOutwardKilled(attachee, args, evt_obj):
+	args.remove_spell()
+	args.remove_spell_mod()
+	return 0
+	
+def MagicCircleOutwardSpellEnd(attachee, args, evt_obj):
+	spell_id = args.get_arg(0)
+	type = args.get_arg(2)
+	if evt_obj.data1 == spell_id:
+		if type == 1:
+			game.particles( 'sp-Magic Circle against Good-END', attachee)
+		elif type == 2:
+			game.particles( 'sp-Magic Circle against Evil-END', attachee)
+		elif type == 3:
+			game.particles( 'sp-Magic Circle against Law-END', attachee)
+		elif type == 4:
+			game.particles( 'sp-Magic Circle against Chaos-END', attachee)
+	return 0
+	
+def HasMagicCircleQuery(attachee, args, evt_obj):
+	type = args.get_arg(2)
+	if evt_obj.data1 == type:
+		evt_obj.return_val = 1
+	return 0
+	
+def MagicCircleOutwardTooltip(attachee, args, evt_obj):
+	type = args.get_arg(2)
+	TypeName = GetSpellType(type)
+	evt_obj.append("Magic Circle Against " + TypeName)
```

Comment:
Should be doable.

Reply:
Which function do you use to pull the value of a spell mesline in python?

Reply:
There isn't any... 
The hacky solution is to add game.get_spell_mesline. 
A better solution is to add game.get_mesline(filepath, line), which would internally convert it to meshandle. To make things more efficient, we should maintain a dictionary that maps mes filename to mes handle and populate it on the go (using MesHandleFind at 0x101e63e0). Since we haven't needed that so far it's not so important, but in my experience you'll happen to have someone clamoring for it soon enough anyway :)

Reply:
Sounds good.  I'll try to add the new function.  It may prove useful in the future.